nadeo_event_api/src/nadeo_event_api/api/structure/settings/script_settings.py

```diff
@@ -25,37 +50,33 @@ def __init__(
         :param warmup_timeout: Time to finish in seconds after the winners, equivalent of finish_timeout but for warmup, only if warmup_duration is -1. -1: Time based on AT (5 sec + AT / 6). Default -1.  
         :param pick_ban_enable: Enable pick and ban. Defining a pick ban order in plugin settings without this enabled will not enable it.
         """
-        self._chat_time = chat_time
-        self._force_laps_number = force_laps_number
-        self._respawn_behavior = respawn_behavior
-        self._warmup_duration = warmup_duration
-        self._warmup_number = warmup_number
-        self._warmup_timeout = warmup_timeout
-        self._pick_ban_enable = pick_ban_enable
+        self.base_script_settings_variables = base_script_settings_variables
+
+        # self._chat_time = base_script_settings_variables.chat_time
+        # self._force_laps_number = force_laps_number
+        # self._respawn_behavior = respawn_behavior
+        # self._warmup_duration = warmup_duration
+        # self._warmup_number = warmup_number
+        # self._warmup_timeout = warmup_timeout
+        # self._pick_ban_enable = pick_ban_enable
 
     def as_jsonable_dict(self) -> dict:
         script_settings = {}
-        script_settings["S_ChatTime"] = self._chat_time
-        script_settings["S_ForceLapsNb"] = self._force_laps_number
-        script_settings["S_RespawnBehaviour"] = self._respawn_behavior.value
-        script_settings["S_WarmUpDuration"] = self._warmup_duration
-        script_settings["S_WarmUpNb"] = self._warmup_number
-        script_settings["S_WarmUpTimeout"] = self._warmup_timeout
-        script_settings["S_PickAndBan_Enable"] = self._pick_ban_enable
+        script_settings["S_ChatTime"] = self.base_script_settings_variables._chat_time
+        script_settings["S_ForceLapsNb"] = self.base_script_settings_variables._force_laps_number
+        script_settings["S_RespawnBehaviour"] = self.base_script_settings_variables._respawn_behavior.value
+        script_settings["S_WarmUpDuration"] = self.base_script_settings_variables._warmup_duration
+        script_settings["S_WarmUpNb"] = self.base_script_settings_variables._warmup_number
+        script_settings["S_WarmUpTimeout"] = self.base_script_settings_variables._warmup_timeout
+        script_settings["S_PickAndBan_Enable"] = self.base_script_settings_variables._pick_ban_enable
 
         return script_settings
 
 
 class ChampionScriptSettings(ScriptSettings):
     def __init__(
         self,
-        chat_time: int = 10,
-        force_laps_number: int = -1,
-        respawn_behavior: RespawnBehavior = RespawnBehavior.DEFAULT,
-        warmup_duration: int = 0,
-        warmup_number: int = 0,
-        warmup_timeout: int = -1,
-        pick_ban_enable: bool = False,
+        base_script_settings_variables: BaseScriptSettings,
```

Comment:
For this and the rest of the ScriptSettings subclasses, include the ` = BaseScriptSettings()` so it's an optional arg with defaults inherited from the class